pkg/costmodel/cluster_helpers.go

```diff
@@ -246,10 +246,10 @@ func buildGPUCostMap(
 		clusterAndNameToType[keyNon] = nodeType
 
 		// If gpu count is available use it to multiply gpu cost
-		if value, ok := gpuCountMap[key]; ok && value != 0 {
+		if value, ok := gpuCountMap[key]; ok {
 			gpuCostMap[key] = gpuCost * value
 		} else {
-			gpuCostMap[key] = gpuCost
+			gpuCostMap[key] = 0
```

Comment:
Is it possible to be in a situation with custom pricing in which:
1. A node's GPU cost is (intentionally) set to something > 0
2. Our GPU count query doesn't pick up the node's GPU, possibly because the node metadata is set up incorrectly or our tracking of node metadata is incomplete

I think this change would cause the node's GPU cost to be 0 when it shouldn't be.

---

But perhaps that's okay and we should just be setting count correctly! If that's the case, then the code change you've made is perfect.

Reply:
That was one of the scenarios I was considering and I think that if we can't/didn't count the GPUs correctly pricing would be wrong anyways. At least if it shows up as a zero and you know the node has a GPU that is a better indication that something is wrong vs showing incorrect pricing on a node that has 10 GPUs and we show the cost of 1.

Reply:
I'm on board with that reasoning.

Reply:
This looks good to me. Let's just make sure that @AjayTripathy is aware.

Reply:
I think this is fine but let's not snap this into the release since I'm not sure of all the implications and we don't have good GPU integration tests...that sounds good to everyone? Let's take a follow on task to add a GPU integration test for GCP and AWS, we'll have a month to get confident here.